server/container/core/src/test/java/org/apache/james/server/core/MailImplTest.java

```diff
@@ -344,7 +342,6 @@ void mailImplShouldBeSerializableWithOptionalAttribute() throws Exception {
             .isEqualToComparingFieldByField(mail);
     }
 
-    @Disabled("JAMES-2665: Throws java.io.NotSerializableException: java.io.NotSerializableException: org.apache.mailet.AttributeValue")
```

Comment:
these 2 cases are not handled by Jsonable? Why? Cannot we modify Jsonable instead of fallback to Java serialization?

Reply:
That's the problem: `AttributeValue does not 'java serialize' well **by design** :trollface: 

So as FileMailRepository relies on MailImpl::serialize then I had to modify the logic here

And finally please not that:
 - This fallback is NOT here to handle issues with Jsonable/AttributeValue`
 - ... but to handle mails serialized BEFORE this fix and thus relying 100% on java serialization for attribute.